tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,276 @@
+		LogLevel:    networkRunnerLogLevel,
+		Endpoint:    gRPCEp,
+		DialTimeout: 10 * time.Second,
+	})
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	ginkgo.By("calling start API via network runner", func() {
+		outf("{{green}}sending 'start' with binary path:{{/}} %q\n", execPath)
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		resp, err := cli.Start(
+			ctx,
+			execPath,
+			client.WithPluginDir(pluginDir),
+			client.WithCustomVMs(map[string]string{
+				"subnetevm": vmGenesisPath,
+			}))
+		cancel()
+		gomega.Expect(err).Should(gomega.BeNil())
+		outf("{{green}}successfully started:{{/}} %+v\n", resp.ClusterInfo.NodeNames)
+	})
+
+	// TODO: network runner health should imply custom VM healthiness
+	// or provide a separate API for custom VM healthiness
+	// "start" is async, so wait some time for cluster health
+	time.Sleep(2 * time.Minute)
+
+	ctx, cancel := createDefaultCtx()
+	_, err = cli.Health(ctx)
+	cancel()
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	subnetEVMRPCEps = make([]string, 0)
+	blockchainID, logsDir := "", ""
+	k, err := getVMID("subnetevm")
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	// wait up to 5-minute for custom VM installation
+	ctx, cancel = context.WithTimeout(context.Background(), 5*time.Minute)
+done:
+	for ctx.Err() == nil {
+		select {
+		case <-ctx.Done():
+			break done
+		case <-time.After(5 * time.Second):
+		}
+
+		outf("{{magenta}}checking custom VM status{{/}}\n")
+		cctx, ccancel := createDefaultCtx()
+		resp, err := cli.Status(cctx)
+		ccancel()
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// all logs are stored under root data dir
+		logsDir = resp.GetClusterInfo().GetRootDataDir()
+
+		if v, ok := resp.ClusterInfo.CustomVms[k.String()]; ok {
+			blockchainID = v.BlockchainId
+			outf("{{blue}}subnet-evm is ready:{{/}} %+v\n", v)
+			break done
+		}
+	}
+	gomega.Expect(ctx.Err()).Should(gomega.BeNil())
+	cancel()
+
+	gomega.Expect(blockchainID).Should(gomega.Not(gomega.BeEmpty()))
+	gomega.Expect(logsDir).Should(gomega.Not(gomega.BeEmpty()))
+
+	cctx, ccancel := createDefaultCtx()
+	uris, err := cli.URIs(cctx)
+	ccancel()
+	gomega.Expect(err).Should(gomega.BeNil())
+	outf("{{blue}}avalanche HTTP RPCs URIs:{{/}} %q\n", uris)
+
+	for _, u := range uris {
+		rpcEP := fmt.Sprintf("%s/ext/bc/%s/rpc", u, blockchainID)
+		subnetEVMRPCEps = append(subnetEVMRPCEps, rpcEP)
+		outf("{{blue}}avalanche subnet-evm RPC:{{/}} %q\n", rpcEP)
+	}
+
+	pid := os.Getpid()
+	outf("{{blue}}{{bold}}writing output %q with PID %d{{/}}\n", outputPath, pid)
+	ci := clusterInfo{
+		URIs:     uris,
+		Endpoint: fmt.Sprintf("/ext/bc/%s", blockchainID),
+		PID:      pid,
+		LogsDir:  logsDir,
+	}
+	gomega.Expect(ci.Save(outputPath)).Should(gomega.BeNil())
+
+	b, err := os.ReadFile(outputPath)
+	gomega.Expect(err).Should(gomega.BeNil())
+	outf("\n{{blue}}$ cat %s:{{/}}\n%s\n", outputPath, string(b))
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	switch mode {
+	case "test":
```

Comment:
can we use constants for `test` / `run` here?

Also, I don't think we should be relying on the test suite as the way to start a local network without shutting it down by passing in a `mode` parameter in this way.

Reply:
Yeah we can remove later when we converge into a single CLI tool. This is to keep the existing feature in runner script :)